ducktape/tests/runner.py

```diff
@@ -260,7 +260,7 @@ def run_all_tests(self):
 
                 if self._expect_client_requests:
                     try:
-                        event = self.receiver.recv(timeout=self.session_context.test_runner_timeout)
+                        event = self.receiver.recv(timeout=-1)
```

Comment:
It feels like we should _still_ have a timeout here, but longer. E.g., 2x as long plus 1 minute compared to the runner timeout. Hitting that timeout would be an unexpected error along the lines of the current error, but would only happen in more exceptional cases as when the client crashed abruptly or some issue occurs which prevents the client message from arriving.